cmd/helm/flags.go

```diff
@@ -55,19 +55,22 @@ func addChartPathOptionsFlags(f *pflag.FlagSet, c *action.ChartPathOptions) {
 // bindOutputFlag will add the output flag to the given command and bind the
 // value to the given format pointer
 func bindOutputFlag(cmd *cobra.Command, varRef *output.Format) {
-	f := cmd.Flags()
-	flag := f.VarPF(newOutputValue(output.Table, varRef), outputFlag, "o",
+	cmd.Flags().VarP(newOutputValue(output.Table, varRef), outputFlag, "o",
 		fmt.Sprintf("prints the output in the specified format. Allowed values: %s", strings.Join(output.Formats(), ", ")))
 
-	completion.RegisterFlagCompletionFunc(flag, func(cmd *cobra.Command, args []string, toComplete string) ([]string, completion.BashCompDirective) {
+	err := cmd.RegisterFlagCompletionFunc(outputFlag, func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
 		var formatNames []string
 		for _, format := range output.Formats() {
 			if strings.HasPrefix(format, toComplete) {
 				formatNames = append(formatNames, format)
 			}
 		}
-		return formatNames, completion.BashCompDirectiveDefault
+		return formatNames, cobra.ShellCompDirectiveDefault
 	})
+
+	if err != nil {
+		log.Fatal(err)
```

Comment:
There are a number of `log.Fatal` calls added to the completion functions.  Could this cause Helm to exit when completion is not critical?  Meaning should we print an error and continue?

Reply:
I wasn't sure if it followed helm's correct practices when I added the `log.Fatal()` calls.

You'll notice that all of them are on the initialization code path, and not on the completion generation path.  Such a `log.Fatal()` call would be triggered when receiving a Cobra error when setting up completion during initialization, and would be caused by a coding error.  So I thought it was ok to abort helm since this would be seen immediately during development.

But I'll follow whichever best practice you recommend.